src/utils/EventEmitter.ts

```diff
@@ -8,24 +8,30 @@ function uuidv4(): string {
 
 /** Generic Event Emitter class following a subscribe/publish paradigm. */
 export class EventEmitter<T extends any[]> {
-  subscribers: Record<string, (...args: [...T]) => void | undefined> = {};
+  private subscribers: Map<string, (...args: [...T]) => void | undefined> = new Map();
+
+  constructor() {}
 
   subscribe(s: (...args: [...T]) => void): () => void {
     let uuid = uuidv4();
-    while (this.subscribers[uuid] !== undefined) uuid = uuidv4();
-    this.subscribers[uuid] = s;
+    while (this.subscribers.get(uuid) !== undefined) uuid = uuidv4();
+    this.subscribers.set(uuid, s);
 
     return () => {
-      delete this.subscribers[uuid];
+      this.subscribers.delete(uuid);
     };
   }
 
   emit(...args: [...T]): void {
-    for (const s in this.subscribers) {
-      const sub = this.subscribers[s];
-      if (sub === undefined) continue;
-
+    this.subscribers.forEach((sub) => {
       sub(...args);
+    });
+  }
+
+  /** Emits the event soon (i.e. sometime shortly after control returns to the browser).  */
+  emitSoon(...args: [...T]): void {
```

Comment:
It looks like this is *not* true in React 17. We can call `ReactDOM.unstable_batchedUpdates` around updates to opt into that behaviour though.

Reply:
OK, I went and read up and learned a bunch more about how React batches updates. Indeed, the batching behavior in 17 (and 16 before it) was more complicated; the official docs glossed over it by simply saying "it batches updates in various circumstances" without going into detail on the circumstances.

But yes, the point of all of this (I assume) is batching the updates; otherwise, there's no reason not to let them continue to run on their own timers. In which case, there's no reason to throw on an extra delay.